File: Semestre_2/Vision_Computadora/Entregas/Matlab/T4_2_Interpolacion.m
clear;
close all;
clc;

[file,path] = uigetfile({'*.*';'*.jpg';'*.png';'*.jpeg';'*.bmp';'*.gif'}, 'File Selector');

ruta = strcat(path,file);
A=imread(ruta);
A=im2double(A);

figure, 
subplot(221), imshow(A);title('Imagen Pequeña');

R1=imresize(A,10,'nearest');
R2=imresize(A,10,'bilinear');
R3=imresize(A,10,'bicubic');

subplot(222), imshow(R1);title('Zoom Vecino Cercano');
subplot(223), imshow(R2);title('Zoom Bilineal');
subplot(224), imshow(R3);title('Zoom Bicubica');

File: Semestre_2/Vision_Computadora/Entregas/Python/T4_2_Interpolacion.py
import matplotlib.pyplot as plt
from skimage.io import imread
from tkinter import Tk, filedialog
from skimage.transform import resize

Tk().withdraw()
file_path = filedialog.askopenfilename(
    title='File Selector',
    filetypes=[("Images", "*.*;*.jpg;*.png;*.jpeg;*.bmp;*.gif")]
)

A = imread(file_path).astype(float) / 255.0
if A.ndim == 3 and A.shape[-1] == 4:
    A = A[..., :3]

fx, fy = A.shape[0]*10, A.shape[1]*10

R1 = resize(A, (fx, fy), order=0, preserve_range=True, anti_aliasing=False)
R2 = resize(A, (fx, fy), order=1, preserve_range=True, anti_aliasing=False)
R3 = resize(A, (fx, fy), order=3, preserve_range=True, anti_aliasing=False)

plt.figure()
plt.subplot(2,2,1); plt.imshow(A);  plt.title('Imagen Pequeña');      plt.axis('off')
plt.subplot(2,2,2); plt.imshow(R1); plt.title('Zoom Vecino Cercano'); plt.axis('off')
plt.subplot(2,2,3); plt.imshow(R2); plt.title('Zoom Bilineal');       plt.axis('off')
plt.subplot(2,2,4); plt.imshow(R3); plt.title('Zoom Bicubica');       plt.axis('off')
plt.show()
